src/webgl/vertex-array.js

```diff
@@ -196,6 +211,7 @@ export default class VertexArray extends Resource {
     layout = layout !== undefined ? layout : buffer.layout;
     assert(target, 'setBuffer needs target');
     assert(layout, 'setBuffer called on uninitialized buffer');
+    this._filledLocations[location] = true;
```

Comment:
Thought: Instead of maintaining a boolean map, you could just have a map of locations to buffers... If there is a buffer, it is `truthy`. Might be more general...